elements/lisk-chain/src/chain.ts

```diff
@@ -433,7 +426,16 @@ export class Chain {
 	public async setValidators(
 		validators: { address: Buffer; isConsensusParticipant: boolean }[],
 		stateStore: StateStore,
+		blockHeader: BlockHeader,
```

Comment:
I believe the chain should be aware o the last block header and validators should be always updated based on last block. So need to pass extra block header param.

Reply:
This `blockHeader` is not the `lastBlock` that chain holds, but is currently being processed block.
Therefore, it needs to be passed to this function.

ex)
this.lastBlock.height === 2 then blockHeader.height will be 3

Reply:
If that's the case always, can't we use that offset ourself in chain? to reduce an extra parameter and give user flexibility to update validators in custom module without need to worry about block header. What do you think?

Reply:
Actually, from the custom module, this parameter is not exposed.
https://github.com/LiskHQ/lisk-sdk/pull/5782/files#diff-b543eea0d99418fe1fb34ff986a2e848R450-R452
Custom module will not need to pass the block header manually